mattermod.go

```diff
@@ -74,5 +71,18 @@ func main() {
 	c.Start()
 	sig := make(chan os.Signal, 1)
 	signal.Notify(sig, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
-	<-sig
+
+	select {
+	case <-sig:
+		mlog.Info("Stopping Mattermod Server")
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		if err := s.Stop(ctx); err != nil {
+			mlog.Error("Error while shutting down server", mlog.Err(err))
+			os.Exit(1)
+		}
```

Comment:
IMO, we can move it inside `s.Stop`, but 0/5. Up to you.

Reply:
I can move the context snippet inside `s.Stop` 👍